lib/session.js

```diff
@@ -414,9 +417,12 @@ Session.prototype._handleDisposition = function(frame) {
     state: frame.state
   };
 
-  _.values(this._linksByName).forEach(function(link) {
-    link._dispositionReceived(disposition);
-  });
+  var dispositionHandler = function(l) { l._dispositionReceived(disposition); };
+  if (frame.role === constants.linkRole.sender) {
+    this._senderLinks.forEach(dispositionHandler);
```

Comment:
I think this is actually reversed - the "role" is the role of the one sending you the dispo frame (so e.g. "I'm a receiver, and I'm sending this dispo frame to the sender").